dateparser/parser.py

```diff
@@ -524,7 +524,7 @@ def parse_alpha(token, skip_component=None):
 class tokenizer(object):
     digits = u'0123456789:'
     letters = u'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'
-    nonwords = u"./\()\"',.;<>~!@#$%^&*|+=[]{}`~?-     "
+    nonwords = u"./\()\"',.;<>~!@#$%^&*|+=[]{}`~?-—–     "
```

Comment:
Thanks @noviluni ! As I understand, this still won't fix cases with other unexpected symbols, e.g. ``dateparser.parse('Oct 1 2018 4:40 PM EST 😊')`` would still fail. I wonder if it's possible to handle "unknown" symbols in another place? But maybe it's a bad idea, I'm not familiar with this part of the code.

Reply:
Hi @lopuhin 

My PR was to fix the problem as soon as possible, as a dash is a common sign, but I have noted this down to investigate it further.

It seems that the tokenizer sepparate between "words" (returns `1`), "digits" (returns `0`, it can include `:`, etc.) and "no-words" (returns `2`) and, after that, it selects the valid tokens by discarding the "no words". However, those unexpected symbols return just an empty string and that breaks the filtering because it compares with 1 and gives the error you had. Fixing this should be as easy as changing this:

```
self.filtered_tokens = [t for t in self.tokens if t[1] <= 1]
```
by this:
```
self.filtered_tokens = [t for t in self.tokens if t[1] != '' and t[1] <= 1]
```

But I'm not sure of the implications and why it was decided to be implemented in this way...

I will do some more tests and push a "fixed" version.

Reply:
> My PR was to fix the problem as soon as possible, as a dash is a common sign, but I have noted this down to investigate it further.

This makes sense, thank you @noviluni :+1:

I checked the code and I think this fix with ``filtered_tokens`` looks good and should not break anything it seems :+1: 

I was also thinking why was this code written like this in the first place, and maybe I have an idea: in python 2 it's not an error to compare an empty string with an int, and ``1 < ''`` is True - so maybe the idea behind a ``''`` default in ``_switch`` method was to have a value which would be greater than any number that would be added next. If this logic is correct, then indeed your fix would be good, and another potential fix would be to default to a large number in ``_switch``.

Reply:
Oh, right, Python 2... It was only 1.5 months ago when it died and I've already forgotten...

If that's the case Instead of filtering the tokens it would be better to return just `3` by default (I mean, for unrecognized characters), as returning the same datatype seems easier to mantain / follow and we will avoid comparing all the valid cases with `""` in the `filtered_tokens` (better performance). However, I think we don't need the `nonwords` list at all, as we could just return 2 when it isn't neither a letter nor a word... 

@lopuhin I have just pushed the new code. Let me know what you think

Reply:
Since `tokenizer` is not a private class (i.e. `_tokenizer`), and these changes change the behavior (for the better) instead of simply fixing the cast bug, I think we might need to deprecate `tokenizer`, copy it as `_tokenizer`, fix `_tokenizer` and use `_tokenizer` in our internal code.

I was worried that these changes would also break our code, as all non-ASCII characters are now considered "non-words", but it looks like in the parts of Dateparser where the tokenizer is used that is not an issue (or our test suite is incomplete around this; let’s hope not).

Reply:
> I was worried that these changes would also break our code, 

I also looked for cases where we would distinguish between non-words and "other" and didn't find any.